saleor/plugins/openid_connect/utils.py

```diff
@@ -245,12 +270,39 @@ def get_user_from_oauth_access_token(
             "Failed to fetch OIDC user info", extra={"user_info_url": user_info_url}
         )
         return None
-    user = get_or_create_user_from_payload(user_info, oauth_url=user_info_url)
-    if not use_scope_permissions:
+    user = get_or_create_user_from_payload(
+        user_info,
+        oauth_url=user_info_url,
+    )
+
+    email_domain = get_domain_from_email(user.email)
+    is_staff_email = email_domain in staff_user_domains
+    if not use_scope_permissions and not is_staff_email:
         user.is_staff = False
+    elif is_staff_email:
+        assign_staff_to_default_group_and_update_permissions(
+            user, staff_default_group_name
+        )
+
     return user
 
 
+def assign_staff_to_default_group_and_update_permissions(
+    user: "User", default_group_name: str
+):
+    """Assign staff user to the default permission group. and update user permissions.
+
+    If the group doesn't exist, the new group without any assigned permissions or
```

Comment:
Are you sure that `get_or_create` is the correct approach here? Consider the following cases:

- What if the provided default group is an empty string?
- What if the provided `default_group_name` has spaces in the name, like `" my-group"`?
- What if someone makes a typo when trying to assign a default group? For example, assuming that you have a group named `OpenID group`, but a staff member provides `openid group` or `OpenID-group`. This would result in the creation of a new group. Is this expected behavior? Shouldn't we raise an exception that group doesn’t exist during the validation of the data to save on the OpenID plugin side?

Reply:
If we would rise an exception in that case, then we should also consider a case when the group has been removed.

Reply:
So for sure I should do the `strip` before, and check if the group name is not empty. Regarding checking if the group exists I can't entirely agree. In the feature, there is an assumption that when the group doesn't exist we should create a new one. It will be mentioned in the docs.

Reply:
I updated the method - added a strip on value and I don't create the group when the value is empty